crates/wasmtime/src/instance.rs

```diff
@@ -145,25 +153,8 @@ impl Instance {
     /// [inst]: https://webassembly.github.io/spec/core/exec/modules.html#exec-instantiation
     /// [issue]: https://github.com/bytecodealliance/wasmtime/issues/727
     /// [`ExternType`]: crate::ExternType
-    pub fn new(module: &Module, imports: &[Extern]) -> Result<Instance, Error> {
+    pub fn new(module: &Module, imports: &[Extern]) -> Result<NewInstance, Error> {
```

Comment:
Is `NewInstance` still necessary here? I'd imagine that 95% of invocations of `Instance::new` will immediately want to `.start()?` aftewards (and seems like 100% in this PR?). Do we have a use case for eventually getting an `Instance` but not running the `start` function?

Reply:
I guess you're right. I was initially thinking of https://github.com/bytecodealliance/lucet/pull/506, but on reflection, I think this will become less important over time. If we ever do need this, we can always add a separate `Instance` constructor to construct a partially-initialized `Instance`.